src/geocel/detail/PolygonUtils.hh

```diff
@@ -0,0 +1,126 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file geocel/detail/LengthUnits.hh
+//! \brief NOTE: only use inside geocel; prefer celeritas/Units.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+//#define CELER_ICRT inline constexpr real_type
+#include <iostream>
+#include <vector>
+#include <corecel/cont/Array.hh>
+#include <corecel/math/ArrayUtils.hh>
+#include <orange/OrangeTypes.hh>
+
+namespace celeritas
+{
+namespace geoutils
+{
+
+template<typename T>
+using Real2 = Array<T, 2>;
+template<typename T>
+using VecReal2 = std::vector<Real2<T>>;
+template<typename T>
+using Real3 = Array<T, 3>;
+template<typename T>
+using VecReal3 = std::vector<Real3<T>>;
+
+enum PgonOrient : int
+{
+    clockwise = -1,
+    collinear = 0,
+    counterclockwise = 1,
+    kCW = clockwise,
+    kLine = collinear,
+    kCCW = counterclockwise
+};
+
+//---------------------------------------------------------------------------//
+// Find orientation of ordered triplet (a, b, c).
+// The function returns following values
+//  0 --> input points are collinear
+// -1 --> Clockwise
+// +1 --> Counterclockwise
+template<typename T>
+CELER_FUNCTION PgonOrient orientation_2D(Real2<T> const& a,
+                                         Real2<T> const& b,
+                                         Real2<T> const& c)
+{
+    T crossp = (b[0] - a[0]) * (c[1] - b[1]) - (b[1] - a[1]) * (c[0] - b[0]);
+    if (crossp == 0)
+        return collinear;  // collinear
+    return (crossp < 0) ? kCW : kCCW;  // clockwise or counterclockwise
+}
+
+//---------------------------------------------------------------------------//
+/*! Check if a 2D polygon is convex.
+ *
+ * /param corners the vertices of the polygon
+ * /param degen_ok if true, consecutive degenerate points are okay, still
+ *   returns true
+ */
+template<typename T>
+CELER_FUNCTION bool
+is_convex_2D(VecReal2<T> const& corners, bool degen_ok = false)
```

Comment:
- Instead of using a `vector<Real2>`, use a `Span<Real2 const>` so we could use other data types (e.g., if we have a fixed-size array)
- The flag here isn't obvious if you only see the calling function (`is_convex(..., true)` means what???) and it's never actually used in the calling code, so just remove it.

Reply:
I've kept the `degen_ok` flag for now, as it is going to be useful when degenerate points are properly treated.